File: 自定义TabBar 适配X OC/YFTabBar/YFTabBar/Controller/CenterViewController.m
//
//  CenterViewController.m
//  YFTabbarController
//
//  Created by student on 2018/3/12.
//  Copyright © 2018年 kangbing. All rights reserved.
//

#import "CenterViewController.h"
#import "YFTabbarController.h"

@interface CenterViewController ()

@end

@implementation CenterViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.title = @"Center";
    
    self.view.backgroundColor = [UIColor whiteColor];
    
    UIButton *btn = [UIButton buttonWithType:UIButtonTypeCustom];
    
    [btn setImage:[UIImage imageNamed:@"plusX_Last"] forState:UIControlStateNormal];
    
    btn.bounds = CGRectMake(0, 0, 64, 64);
    
    btn.center = self.view.center;
    
    [self.view addSubview:btn];
    

        
    [btn addTarget:self action:@selector(close) forControlEvents:UIControlEventTouchUpInside];

    
    
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


-(void)close{
    
  
    [self dismissViewControllerAnimated:YES completion:nil];
    
}

@end

File: 自定义TabBar 适配X  Swift/YFTabBar/YFTabBarController/Classes/Controllers/CenterViewController.swift
//
//  CenterViewController.swift
//  YFTabBarController
//
//  Created by yangfan on 2018/3/15.
//  Copyright © 2018年 yangfan. All rights reserved.
//

import UIKit

class CenterViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        
        view.backgroundColor = UIColor.purple
        
        addOneBtn()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}

extension CenterViewController{
    
    
    func addOneBtn(){
        
        let btn = UIButton(type: UIButtonType.custom)
        
        btn.setImage(UIImage(named:"plusX_Last"), for: UIControlState.normal)
        
        btn.bounds = CGRect(x: 0, y: 0, width: 64, height: 64)
        
        btn.center = view.center
        
        btn.addTarget(self, action:#selector(close) , for: UIControlEvents.touchUpInside)
        
        view.addSubview(btn)
    }
    
    @objc func close(){
        
        dismiss(animated: true, completion: nil)
        
    }
    
    
}
